### PR TITLE
Add moving mesh example

### PR DESCRIPTION
**Pull request by alexfikl**
Adds a simple example with a moving mesh to track some progress in #123.

### CODE REVIEW

File: meshmode/discretization/poly_element.py [hunk 1]
```diff
@@ -234,6 +234,11 @@ def weights(self):
                 np.ones(len(self.basis())))
 
 
+@memoize
+def _warp_and_blend_nodes(dim, order):
+    return mp.warp_and_blend_nodes(dim, order)
```
Comment (alexfikl) [inline meshmode/discretization/poly_element.py:239]: Caching the unit nodes here seems to be the biggest added gain. This is hack at the moment, but would be nice to figure something out.

Here's some profiling logs for with/without (both are html files)
[pyinstrument.zip](https://github.com/inducer/meshmode/files/6184163/pyinstrument.zip)

File: meshmode/discretization/__init__.py [hunk 2]
```diff
@@ -368,7 +368,9 @@ def prg():
 
         def resample_mesh_nodes(grp, iaxis):
             # TODO: would be nice to have the mesh use an array context already
-            nodes = actx.from_numpy(grp.mesh_el_group.nodes[iaxis])
+            nodes = grp.mesh_el_group.nodes[iaxis]
+            if isinstance(nodes, np.ndarray):
+                nodes = actx.freeze(actx.from_numpy(nodes))
```
Comment (inducer) [inline meshmode/discretization/__init__.py:373]: What other type is being accepted here? Could you make that explicit? I find the code hard to read without that.

**Review by inducer — COMMENTED** [review]
Thanks! One question about types, otherwise good to go.

File: meshmode/discretization/__init__.py (hunk 2 above)
Comment (alexfikl) [reply meshmode/discretization/__init__.py]: I hacked together a `mesh_to_device` function while profiling a bit to see what that would save. This is just some remnant from that, so I went ahead and removed it.

File: meshmode/discretization/poly_element.py (hunk 1 above)
Comment (alexfikl) [reply meshmode/discretization/poly_element.py:239]: @inducer Before merging, this should be removed too and leave it for another PR. 

Maybe figure out some more general caching for the unit nodes? Although there aren't many objects that survive when moving the discretization..

Comment (alexfikl) [reply meshmode/discretization/poly_element.py]: Removed in 3af9d7d

**Comment by inducer** [comment]
Thanks!

**Pull request merged**